### PR TITLE
[MAINT] Fixed issue #877

### PR DESCRIPTION
**Pull request by Yuriyzabegaev**
## Proposed changes

This fixes the issue #877. I unified return types for the methods `FractureNetwork3d.impose_external_boundary` and `FractureNetwork2d.impose_external_boundary`, and updated a test for the changed return value in 3D. I also tried to improve some docstrings and warnings. 
Note: I did not try to investigate the performance issue that @keileg mentioned in the issue discussion, but we can discuss it.

## Types of changes

What types of changes does this PR introduce to PorePy?
_Put an `x` in the boxes that apply._

- [ ] Minor change (e.g., dependency bumps, broken links).
- [x] Bugfix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [x] Testing (contribution related to testing of existing or new functionality).
- [x] Documentation (contribution related to adding, improving, or fixing documentation).
- [ ] Maintenance (e.g., improve logic and performance, remove obsolete code).
- [ ] Other:

## Checklist

_Put an `x` in the boxes that apply or explain briefly why the box is not relevant._

- [x] The documentation is up-to-date.
- [x] Static typing is included in the update.
- [x] This PR does not duplicate existing functionality.
- [x] The update is covered by the test suite (including tests added in the PR).
- [ ] If new skipped tests have been introduced in this PR, `pytest` was run with the `--run-skipped` flag.

### CODE REVIEW

**Comment by IvarStefansson** [comment]
I tried running the MWE in the issue. As far as I can tell, the warning is indeed no longer raised. However, I fail to understand which changes to the code made the difference. Could you please explain, @Yuriyzabegaev?

File: src/porepy/grids/mdg_generation.py [hunk 1]
```diff
@@ -401,14 +401,12 @@ def _validate_args(
...
-            _, edges_deleted = fracture_network.impose_external_boundary(domain)
-            sz = edges_deleted.size
+            _, fractures_deleted = fracture_network.impose_external_boundary(domain)
         elif dim == 3:
             assert isinstance(fracture_network, FractureNetwork3d)
-            fractures_deleted = fracture_network.impose_external_boundary(domain)
-            # Take note of deleted fractures
-            sz = fractures_deleted.size
-        if sz > 0:
+            _, fractures_deleted = fracture_network.impose_external_boundary(domain)
```
Comment (Yuriyzabegaev) [inline src/porepy/grids/mdg_generation.py:407]: Here, we access the new return data. Previously, the saved fractures mistakenly were assigned to a variable `fractures_deleted` here.

File: src/porepy/fracs/fracture_network_3d.py [hunk 2]
```diff
@@ -2010,7 +2018,7 @@ def new_pairs_and_angles(ei):
         self.tags["boundary"] = boundary_tags
 
         self._reindex_fractures()
-        return ind_map
+        return ind_map, delete_frac
```
Comment (Yuriyzabegaev) [inline src/porepy/fracs/fracture_network_3d.py:2021]: Here, we return not only the kept fractures but also the deleted ones.

File: src/porepy/grids/mdg_generation.py [hunk 3]
```diff
@@ -401,14 +401,12 @@ def _validate_args(
...
+        # Take note of deleted fractures
+        if (sz := fractures_deleted.size) > 0:
```
Comment (OmarDuran) [inline src/porepy/grids/mdg_generation.py:409]: Nice!

**Comment by OmarDuran** [comment]
Thanks a lot @Yuriyzabegaev! for the unification. In my opinion, the performance issue should be addressed separately.

**Review by OmarDuran — APPROVED** [review]
Nice work!

**Review by keileg — COMMENTED** [review]
Looks fine to me.

**Comment by keileg** [comment]
@Yuriyzabegaev Feel free to merge

**Pull request merged**